cmd/lakectl/cmd/validargs.go

```diff
@@ -0,0 +1,59 @@
+package cmd
+
+import (
+	"context"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/treeverse/lakefs/pkg/api"
+	"github.com/treeverse/lakefs/pkg/uri"
+)
+
+func ValidArgsRepository(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	if len(args) != 0 {
+		return nil, cobra.ShellCompDirectiveNoFileComp
+	}
+	return validRepositoryToComplete(cmd.Context(), toComplete)
+}
+
+func validRepositoryToComplete(ctx context.Context, toComplete string) ([]string, cobra.ShellCompDirective) {
+	// do not suggest in case we are passed the repository part
+	uriPrefix := uri.LakeFSSchema + uri.LakeFSSchemaSeparator
+	if strings.HasPrefix(toComplete, uriPrefix) && strings.Contains(toComplete[len(uriPrefix):], uri.PathSeparator) {
+		return nil, cobra.ShellCompDirectiveNoFileComp | cobra.ShellCompDirectiveNoSpace
+	}
+
+	// extract repository name written so far
+	var prefix api.PaginationPrefix
+	if strings.HasPrefix(toComplete, uriPrefix) {
```

Comment:
If it doesn't have the `lakefs://` prefix, we still want an autocomplete, no?

So maybe the prefix should be "toComplete".
For example, if I'm doing `lakectl branch create my-rep<TAB>`, I want to get: `lakectl branch create lakefs://my-repo`

Reply:
Wanted to give the same autocomplete behaviour we have with other application - when the prefix you enter doesn't match any of the possible options we do not suggest completion.
Example: docker autocompete will match the prefix to the image name - but if you will find me a other working reference I will consider :)